pkg/logs/launchers/container/tailerfactory/file.go

```diff
@@ -120,10 +117,13 @@ func (tf *factory) makeDockerFileSource(source *sources.LogSource) (*sources.Log
 	f.Close()
 
 	sourceName, serviceName := tf.defaultSourceAndService(source, containersorpods.LogContainers)
-
+	configType := source.Config.Type
+	if source.Config.Type == "docker" || source.Config.Type == "podman" || source.Config.Type == "containerd" {
```

Comment:
Since we continue to add support for multiple container runtimes maybe inverting this logic would be more maintainable (and possibly safer)? 

ex: 
```
configType := config.FileType
if source.Config.Type == "udp" || source.Config.Type == "tcp" {
   configType = source.Config.Type 
}
```

This way it's always `file` type unless explicitly set to tcp/udp rather than only being `file` type when a specific container runtime is detected. 

(Same for `makeK8sFileOrNetworkProtocolSource` below)